bc_obps/registration/api/_operations/operation_id.py

```diff
@@ -37,157 +28,17 @@
     url_name="get_operation",
 )
 @authorize(AppRole.get_all_authorized_app_roles(), UserOperator.get_all_industry_user_operator_roles())
+@handle_http_errors()
 def get_operation(request, operation_id: UUID):
-    # In this endpoint we are using different schema and different operation queries to optimize the response based on the user role
-    user: User = request.current_user
-    try:
-        operation = (
-            Operation.objects.only(
-                *OperationOut.Config.model_fields,
-                "naics_code",
-                "point_of_contact__address",
-                "point_of_contact__first_name",
-                "point_of_contact__last_name",
-                "point_of_contact__email",
-                "point_of_contact__position_title",
-                "point_of_contact__phone_number",
-                "bc_obps_regulated_operation__id",
-                "operator__physical_address",
-                "operator__mailing_address",
-                "operator__legal_name",
-                "operator__trade_name",
-                "operator__cra_business_number",
-                "operator__bc_corporate_registry_number",
-                "operator__business_structure",
-                "operator__website",
-            )
-            .select_related(
-                "operator__physical_address", "operator__mailing_address", "point_of_contact__address", "naics_code"
-            )
-            .prefetch_related("operator__parent_operators", "regulated_products")
-            .get(id=operation_id)
-        )
-    except Operation.DoesNotExist:
-        raise HttpError(404, "Not Found")
-    if user.is_industry_user():
-        if not operation.user_has_access(user.user_guid):
-            raise HttpError(401, UNAUTHORIZED_MESSAGE)
-        return 200, operation
-    return 200, operation
+    return 200, OperationService.get_operation_if_authorized(get_current_user(request), operation_id)
 
 
 @router.put(
     "/operations/{operation_id}", response={200: OperationUpdateOut, codes_4xx: Message}, url_name="update_operation"
```

Comment:
In the silk profiler, I went from 11 to 13 queries--do you think that's ok until we get to the optimization ticket, or should I work on it in this PR?

Reply:
That is fine for now! After all the API work, we have changed many of the optimized endpoints, but it's worth it for code readability. We can deal with any issues later.